apps/spark-3.5/src/test/java/com/linkedin/openhouse/catalog/e2e/MinimalSparkMoRTest.java

```diff
@@ -0,0 +1,139 @@
+
+  final BiFunction<Operations, Table, RewriteDataFiles.Result> rewriteFunc =
+      (ops, table) ->
+          ops.rewriteDataFiles(
+              table,
+              1024 * 1024, // 1MB
+              1024, // 1KB
+              1024 * 1024 * 2, // 2MB
+              2,
+              1,
+              true,
+              10);
+
+  static Operations ops;
+
+  @BeforeAll
+  public static void startUp() throws Exception {
+    ops = Operations.withCatalog(getSparkSession(), null);
+  }
+
+  @AfterAll
+  public static void shutdown() throws Exception {
+    ops.close();
+  }
+
+  protected List<Object[]> rowsToJava(List<Row> rows) {
+    return rows.stream().map(this::toJava).collect(Collectors.toList());
+  }
+
+  private Object[] toJava(Row row) {
+    return IntStream.range(0, row.size())
+        .mapToObj(
+            pos -> {
+              if (row.isNullAt(pos)) {
+                return null;
+              }
+
+              Object value = row.get(pos);
+              if (value instanceof Row) {
+                return toJava((Row) value);
+              } else if (value instanceof scala.collection.Seq) {
+                return row.getList(pos);
+              } else if (value instanceof scala.collection.Map) {
+                return row.getJavaMap(pos);
+              } else {
+                return value;
+              }
+            })
+        .toArray(Object[]::new);
+  }
+
+  protected List<Object[]> sql(String query, Object... args) {
+    List<Row> rows = ops.spark().sql(String.format(query, args)).collectAsList();
+    if (rows.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    return rowsToJava(rows);
+  }
+
+  protected Object[] row(Object... values) {
+    return values;
+  }
+
+  protected void initTable() {
+    sql(
+        "ALTER TABLE %s SET TBLPROPERTIES ('write.delete.mode'='merge-on-read', 'write.update.mode'='merge-on-read', 'write.merge.mode'='merge-on-read', 'write.delete.distribution-mode'='range');",
+        tableName);
+  }
+
+  protected void createAndInitTable(String schema) {
+    sql("CREATE TABLE openhouse.%s (%s) USING iceberg %s", tableName, schema, "");
+    initTable();
+  }
+
+  @Test
+  public void testDeleteFilesCanBeCreated() throws NoSuchTableException {
+    createAndInitTable("id int, data string");
+
+    List<SimpleRecord> records =
+        Arrays.asList(
+            new SimpleRecord(1, "a"),
+            new SimpleRecord(1, "b"),
+            new SimpleRecord(1, "c"),
+            new SimpleRecord(2, "d"),
+            new SimpleRecord(2, "e"));
+    ops.spark()
+        .createDataset(records, Encoders.bean(SimpleRecord.class))
+        .coalesce(1)
+        .writeTo(tableName)
+        .append();
+
+    sql("DELETE FROM %s WHERE id = 1 and data='a'", tableName);
+    sql("DELETE FROM %s WHERE id = 2 and data='d'", tableName);
+    sql("DELETE FROM %s WHERE id = 1 and data='c'", tableName);
+
+    // TODO: put this in the tablestats API
```

Comment:
what does this todo mean?

Reply:
elsewhere in compaction test code is assertions based on the tableStats utility method.

since we plan to use tableStats as the monitoring method for delete file accumulation, i will do the same thing in these unittests. knocking two birds with one stone (validating delete file feature AND implementing the monitoring for it)